### PR TITLE
Ownerships

### PR DESCRIPTION
**Pull request by BakkerTom**
This adds the concept of ownership to the api and the corresponding UI for it.

### CODE REVIEW

File: src/main/frontend/src/components/cars/TransferOwnershipModal.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    .then(results => {
+                        loading(false);
+                        this.options = results;
+                    })
+
+            },
+
+            didPressSubmit() {
+                if (this.selected) {
+                    console.log('-- ON SUBMIT', this.selected);
```
Comment (ghost) [inline src/main/frontend/src/components/cars/TransferOwnershipModal.vue]: `console.log` statement is superfluous.

File: src/main/frontend/src/services/ownerships.service.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    return fetch(`/government/api/vehicles/${registrationID}/ownerships`, requestOptions)
+        .then(handleResponse)
+        .then(ownerships => {
+            return ownerships;
+        });
+}
+
+function transfer(registrationID, user_id) {
+    console.log(user_id);
```
Comment (ghost) [inline src/main/frontend/src/services/ownerships.service.js]: `console.log` is superfluous.

File: src/main/frontend/src/store/modules/ownerships.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            .then(ownerships => {
+                commit("OWNERSHIPS_SUCCESS", ownerships);
+            })
+            .catch(() => {
+                commit("OWNERSHIPS_FAILURE");
+            })
+    },
+
+    transfer({commit, dispatch}, { registrationID, userID }) {
+        console.log('--VUE ACTION', {registrationID, userID});
```
Comment (ghost) [inline src/main/frontend/src/store/modules/ownerships.js]: `console.log` is superfluous.

File: src/main/frontend/src/store/modules/ownerships.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+const state = {
+    all: [],
+    status: {},
+    transfer_status: {}
+};
+
+const actions = {
+    getOwnerships({commit}, registrationID) {
+        commit("OWNERSHIPS_REQUEST");
```
Comment (ghost) [inline src/main/frontend/src/store/modules/ownerships.js]: possibly make static variable?

File: src/main/java/government/Seeds.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Inject
+    UserFacade userFacade;
+
+    @Inject
+    OwnershipRepository ownershipRepository;
+
+    @PostConstruct
+    private void init() {
+        System.out.println("STARTED SEEDS");
```
Comment (ghost) [inline src/main/java/government/Seeds.java:33]: Possibly introduce logging service.

File: src/main/java/government/Seeds.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        cal.setTime(new Date());
+        cal.add(Calendar.DATE, -10);
+
+        Ownership pastOwnership = new Ownership();
+        pastOwnership.setUser(tim);
+        pastOwnership.setVehicle(peugeot);
+        pastOwnership.setEndDate(cal.getTime());
+
+        ownershipRepository.save(ownership);
+        ownershipRepository.save(ownership1);
```
Comment (ghost) [inline src/main/java/government/Seeds.java:85]: variable naming is vague. Either use `ownership1` and `ownership2` or something like `ownership` and `otherOwnership`.

File: src/main/java/government/controller/VehiclesController.java [hunk 7]
```diff
@@ -1,36 +1,46 @@
...
 import java.util.List;
 import java.util.Optional;
 
 @Path("/vehicles")
 @Produces("application/json")
+@Secured({Role.Employee, Role.Admin})
 public class VehiclesController {
 
     @Inject
     VehicleFacade facade;
```
Comment (ghost) [inline src/main/java/government/controller/VehiclesController.java:30]: should be named `vehicleFacade`.

File: src/main/java/government/controller/VehiclesController.java [hunk 8]
```diff
@@ -1,36 +1,46 @@
...
 
+    @Inject
+    OwnershipFacade ownershipFacade;
+
+    @Inject
+    UserFacade userFacade;
+
     @Inject
     VehicleMapper vehicleMapper;
```
Comment (ghost) [inline src/main/java/government/controller/VehiclesController.java:39]: Why is the `VehicleMapper` not part of the `VehicleFacade`? Shouldn't the mapping be done in the facade? Same for `TrackerIdMapper`.

File: src/main/java/government/facade/UserFacade.java [hunk 9]
```diff
@@ -34,6 +40,10 @@
         return null;
```
Comment (ghost) [inline src/main/java/government/facade/UserFacade.java:40]: `return null` isn't unexpected and undocumented behaviour.

File: src/main/java/government/filter/AuthenticationFilter.java [hunk 10]
```diff
@@ -66,7 +66,7 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
 
             // Validate the token
             String email = jwt.validateToken(token);
-            validateToken(token);
+//            validateToken(token);
```
Comment (ghost) [inline src/main/java/government/filter/AuthenticationFilter.java:69]: commented out code.

File: src/main/java/government/filter/AuthenticationFilter.java [hunk 11]
```diff
@@ -96,29 +96,29 @@ private void abortWithUnauthorized(ContainerRequestContext requestContext) {
...
-            Role roleUser = Role.valueOf(roles);
-
-            List<Role> methodRoles = extractRoles(method);
-
-            if (!methodRoles.contains(roleUser)) {
-                throw new Exception("no valid Role.");
-            }
-        }
-    }
+//    private void validateToken(String token) throws Exception {
```
Comment (ghost) [inline src/main/java/government/filter/AuthenticationFilter.java:99]: LOTS of commented out code.

File: src/main/java/government/model/User.java [hunk 12]
```diff
@@ -21,6 +23,9 @@
     @NotNull
     private String password;
 
+    @OneToMany(mappedBy = "user")
+    private List<Ownership> ownerships;
```
Comment (ghost) [inline src/main/java/government/model/User.java:27]: Is `List` the most basic interface you need for this class? Is `Collection` not sufficient?

File: src/main/java/government/repository/UserRepository.java [hunk 13]
```diff
@@ -28,4 +29,11 @@ private void init() {
             return Optional.empty();
         }
     }
+
+    public List<User> search(String searchTerm) {
+        Query query = entityManager.createQuery("select u from User u where u.name like :term or u.email like :term");
+        query.setParameter("term", "%" + searchTerm + "%");
```
Comment (ghost) [inline src/main/java/government/repository/UserRepository.java:35]: should the wildcard operators be added here? What if you want an exact search?

**Review by OlafvanDijk — APPROVED** [review]

**Pull request merged**